File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # approach: two pointers
        
        l = 0
        r = len(numbers) - 1
        
        while l < r:
            left = numbers[l]
            right = numbers[r]
            
            total = left + right
            
            if total < target:
                l += 1
            elif total > target:
                r -= 1
            else:
                return [l + 1, r + 1]

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // approaach: two pointers
        
        int l = 0;
        int r = numbers.length - 1;
        
        while (l < r) {
            int sum = numbers[l] + numbers[r];
            
            if (sum < target) {
                l++;
            } else if (sum > target) {
                r--;
            } else {
                return new int[] { l + 1, r + 1 };
            }
        }
        
        return null;
    }
}